### PR TITLE
Updated

### PR DESCRIPTION
**Pull request by shirshakk**
[Design: UI revamp of search, contributors and about-us tabs](https://github.com/IEEE-VIT/hacktoberfest-flutter/issues/43)

### Changes made

-   make changes on contribution page , home_page and info_page
-   improve the ui and interconnectivity of the page

### How does the solution address the problem

This PR will give the information about all the contributor , provide the feature to the user to input the new contributor . In aboutUs page it will give the information about the hacktoberfest when clicked on the image and also provide the information IEEE information in the same way

### CODE REVIEW

**Comment by dk-a-dev** [comment]
@shirshakk we have added flutter lint rules with a new commit, please pull that commit and make sure to optimise and refactor your code before putting up any commits as per lint rules, your code editor will help u out in code improvement, by giving warnings make sure to fix them

File: lib/main.dart
Comment (dk-a-dev) [inline lib/main.dart:1]: Splash screen ??
Remove the splash screen, this issue does not intend to add or change the splash screen, also a dedicated screen(file) for splash is there, please stick to dedicated requirement

File: pubspec.yaml [hunk 1]
```diff
@@ -66,7 +66,12 @@ flutter:
 
   # To add assets to your application, add an assets section, like this:
   assets:
-    - assets/
```
Comment (dk-a-dev) [inline pubspec.yaml:69]: this is fine no need to put individual files names here, just putting up folder path will be enough

File: lib/screens/contributors.dart
Comment (dk-a-dev) [inline lib/screens/contributors.dart:1]: Good job fixing the contributors page but, I suggest you upgrade Flutter version and also adopt good practices, like adding the const keyword wherever required
but first things first "flutter upgrade" do an upgrade

File: lib/screens/contributors.dart [hunk 2]
```diff
@@ -295,16 +356,50 @@ Future<List<ContributorCard>> getContributors(
...
+    // Handle the case where GitHub data could not be retrieved
+    print('Failed to fetch GitHub data for the contributor.');
+  }
 }
 
 _launchURL(String gurl) async {
   String url = gurl;
-  if (await canLaunchUrl(Uri.parse(url))) {
-    await launchUrl(Uri.parse(url));
+  if (await canLaunch(url)) {
```
Comment (dk-a-dev) [inline lib/screens/contributors.dart:401]: image

Please stick to newer methods and functions

File: lib/screens/contributors.dart [hunk 3]
```diff
@@ -11,7 +11,7 @@ import 'package:hacktoberfest_flutter/models/contributors_card_model.dart';
 class Contributors extends StatefulWidget {
   static const String routename = '/Contributors';
 
-  const Contributors({super.key});
+  const Contributors({Key? key});
```
Comment (dk-a-dev) [inline lib/screens/contributors.dart:14]: image

Fix: refer left comment

File: lib/screens/contributors.dart [hunk 4]
```diff
@@ -11,7 +11,7 @@ import 'package:hacktoberfest_flutter/models/contributors_card_model.dart';
 class Contributors extends StatefulWidget {
   static const String routename = '/Contributors';
 
-  const Contributors({super.key});
```
Comment (dk-a-dev) [inline lib/screens/contributors.dart:14]: > const Contributors({super.key});

this is absolutely modern way to do this using super keyword
again migrate to newer versions of flutter and dart

File: lib/screens/contributors.dart [hunk 5]
```diff
@@ -60,206 +66,258 @@ class _ContributorsState extends State<Contributors>
...
 
+  void _showAddContributorDialog(
+      BuildContext context, Function(ContributorCard) addContributorCallback) {
+    TextEditingController usernameController = TextEditingController();
+
+    showDialog(
+      context: context,
+      builder: (context) {
+        return AlertDialog(
+          title: Text('Add Contributor'),
```
Comment (dk-a-dev) [inline lib/screens/contributors.dart:286]: image

use const keyword for every widget which avoids unnecessary rebuilds

File: lib/screens/contributors.dart [hunk 6]
```diff
@@ -23,24 +23,30 @@ class _ContributorsState extends State<Contributors>
...
+
+  void addContributor(ContributorCard newContributor) {
+    setState(() {
+      cardList.add(newContributor);
     });
   }
 
   @override
-  // ignore: must_call_super
   Widget build(BuildContext context) {
```
Comment (dk-a-dev) [inline lib/screens/contributors.dart:44]: This method overrides a method annotated as '@mustCallSuper' in 'AutomaticKeepAliveClientMixin', but doesn't invoke the overridden method.

fix: super.build(context); after line 44

File: lib/screens/contributors.dart [hunk 7]
```diff
@@ -295,16 +356,50 @@ Future<List<ContributorCard>> getContributors(
...
+      displayImgUrl: contributorData.avatarUrl,
+      website: contributorData.blog.isEmpty
+          ? contributorData.htmlUrl
+          : contributorData.blog,
+    );
+
+    // Access cardList and update it within the _ContributorsState instance
+    addContributorCallback(newContributor);
+
+    try {
```
Comment (dk-a-dev) [inline lib/screens/contributors.dart:387]: Using BuildContexts across async functions is bad practice see if u can refactor code and avoid it

File: lib/screens/contributors.dart
Comment (dk-a-dev) [reply lib/screens/contributors.dart:1]: i have stated a lot of changes line by line please re check them all and also pull from main branch some new lints are added to enforce good coding practices

File: lib/screens/home_page.dart
Comment (dk-a-dev) [inline lib/screens/home_page.dart:1]: I expected that u would take inspiration from the colour scheme of the hacktoberfest website...
that purpleaccent on one home and blue on contributors and then again purple on info page makes it look bad

Changes Needed:
- add cont wherever necessary
- use hacktoberfest designs and colour palette to make UI changes
- Look the app at light mode and u will realise that white text colour on selection of a tab makes it look disappear please fix the colour scheme
- also change the appbar to new material app bar as its full revamp

Comment (dk-a-dev) [inline lib/screens/home_page.dart:1]: Import on line 2-3 seems like unused imports added by code editor please fix them

File: lib/screens/info_page.dart [hunk 8]
```diff
@@ -16,57 +13,63 @@ class InfoPage extends StatefulWidget {
 
 class _InfoPageState extends State<InfoPage> {
   static const color = Color(0xff13253d);
```
Comment (dk-a-dev) [inline lib/screens/info_page.dart:15]: seems unused

File: lib/screens/info_page.dart
Comment (dk-a-dev) [inline lib/screens/info_page.dart:1]: - add const wherever required
- remove unused variables
- and try to improve UI of info page, try a different design as hacktoberfest image on a purple background looks bad...try taking inspiration from hacktoberfest website or some other platforms like dribble, pintrest, don't forget to stick to theme of hacktoberfesr(colours,fonts etc)

File: lib/screens/search_repos.dart [hunk 9]
```diff
@@ -10,23 +10,17 @@ import '../widgets/new_repo_card.dart';
...
   @override
   State<Search> createState() => _SearchState();
 }
 
 class _SearchState extends State<Search> {
-  final String searchText = '';
-
-  Future<List> listOfRepos = Future.value([]);
-
   TextEditingController _controller = TextEditingController();
```
Comment (dk-a-dev) [inline lib/screens/search_repos.dart:20]: add final keyword

File: lib/screens/search_repos.dart
Comment (dk-a-dev) [inline lib/screens/search_repos.dart:1]: Add const wherever required

File: lib/screens/search_repos.dart [hunk 10]
```diff
@@ -135,18 +133,22 @@ class _SearchState extends State<Search> {
                   borderRadius: BorderRadius.circular(30.0),
                 ),
                 padding: const EdgeInsets.all(1.0),
+                primary: Colors.blue, // Change button background color
```
Comment (dk-a-dev) [inline lib/screens/search_repos.dart:136]: image

Comment (dk-a-dev) [reply lib/screens/search_repos.dart:136]: backgroundColor: Colors.blue,

File: lib/main.dart [hunk 11]
```diff
@@ -1,38 +1,37 @@
...
 import 'package:hacktoberfest_flutter/screens/splash_screen.dart';
+import 'package:animated_splash_screen/animated_splash_screen.dart';
 import 'routes.dart';
 
 void main() {
   runApp(const MyApp());
 }
 
 class MyApp extends StatelessWidget {
-  const MyApp({super.key});
```
Comment (dk-a-dev) [inline lib/main.dart:12]: this is fine, its modern way of passing keys refer concept of parent classes in dart
and modify it back to this state

**Review by dk-a-dev — CHANGES_REQUESTED** [review]
Good Work, but there are  a lot of changes to do please look into in b/w comments
Summary of Review:
- adopt new flutter practices
- add const wherever necessary
- remove unused code parts
- check your flutter version and work on a new one 
- use new functions and methods only avoid deprecated ones
- a commit adding linter rules is added make sure to refactor code properly as per those rules
- major UI changes required, like change in colour palette and theme also in design of info page along with of navigation rail at bottom and also of app bar on top

Please make changes

File: lib/main.dart
Comment (shirshakk) [reply lib/main.dart:1]: in this I have  to remove the github logo and apply the hacktoberfest image?

Comment (dk-a-dev) [reply lib/main.dart:1]: no just remove everything related to splash screen and keep pulling from master to keep ur project fork updated

**Comment by dk-a-dev** [comment]
Hey @shirshakk i am happy to see you fixing things and making changes, but please improve your commit message and commit frequently for better commit history
refer- [conventional commit message standard](https://gist.github.com/qoomon/5dfcdf8eec66a051ecd85625518cfd13)

**Comment by shirshakk** [comment]
I have made all the changes as per the requirements. on [Updated(lib/main.dart:) , please let me know if any more changes are needed to be done

**Comment by shirshakk** [comment]
Hello, I've completed my work. Could you please review it and let me know if any further changes or feedback are needed? Thanks!

**Comment by dk-a-dev** [comment]
@shirshakk , can u also resolve merge conflicts, if u cant le me know I will help

**Comment by shirshakk** [comment]
need help to resolve merge conflicts.

**Comment by shirshakk** [comment]
@dk-a-dev , Need help to resolve merge conflicts

**Comment by dk-a-dev** [comment]
@shirshakk yup k don't worry I will resolve them and then review your pr, also do me a favour please pull from master and make sure optimization and refactoring is done

**Comment by shirshakk** [comment]
please add hactoberfest and hactoberfest-accepted tag

**Comment by dk-a-dev** [comment]
@shirshakk where are all ur commits and why us closed the PR

**Comment by shirshakk** [comment]
I have not closed the PR . This is automatically closed when you resolve the merge conflict

**Comment by shirshakk** [comment]
@dk-a-dev I have not closed the PR

**Comment by dk-a-dev** [comment]
> I have not closed the PR . This is automatically closed when you resolve the merge conflict

No buddy how is that possible, a lot of people resolved conflicts and asked to me to merge it
something went wrong from ur end

**Comment by shirshakk** [comment]
Please guide me what to do now

**Comment by shirshakk** [comment]
What if i create a new PR with changes?

**Comment by dk-a-dev** [comment]
@shirshakk please give progress report

**Comment by dk-a-dev** [comment]
@shirshakk your conflicts are unsolvable and creating problems in master
I suggest refiling a PR on the issue, recloning would be great as this will update ur branch with master

**Pull request closed without merging**